forge-gui/res/cardsfolder/upcoming/dreams_of_steel_and_oil.txt

```diff
@@ -0,0 +1,9 @@
+Name:Dreams of Steel and Oil
+ManaCost:B
+Types:Sorcery
+A:SP$ RevealHand | ValidTgts$ Opponent | ImprintRevealed$ True | RememberTargets$ True | SubAbility$ DBChooseCard | SpellDescription$ Target opponent reveals their hand. You choose an artifact or creature card from it, then choose an artifact or creature card from their graveyard. Exile the chosen cards.
+SVar:DBChooseCard:DB$ ChooseCard | ChoiceZone$ Hand | Choices$ Artifact.IsImprinted,Creature.IsImprinted | RememberChosen$ True |  ChoiceTitle$ Choose an artifact or creature card | Mandatory$ True | SubAbility$ DBChooseFromGraveyard
+SVar:DBChooseFromGraveyard:DB$ ChooseCard | ChoiceZone$ Graveyard | Choices$ Artifact.RememberedPlayerOwn,Creature.RememberedPlayerOwn | RememberChosen True | ChoiceTitle$ Choose an artifact or creature card | Mandatory$ True | SubAbility$ DBExileBoth
+SVar:DBExileBoth:DB$ ChangeZone | Defined$ RememberedCard | Origin$ Graveyard | Origin$ Hand | Destination$ Exile | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True | ClearImprinted$ True 
```

Comment:
you'll also need to ClearChosenCard

Reply:
@Northmoc 
that information is not copied when card changes zone